File: app-android/app/src/main/java/de/tutao/tutanota/data/AlarmInfoDao.java
package de.tutao.tutanota.data;

import static androidx.room.OnConflictStrategy.REPLACE;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

import de.tutao.tutanota.alarms.AlarmNotification;

@Dao
public interface AlarmInfoDao {
	@Insert(onConflict = REPLACE)
	void insertAlarmNotification(AlarmNotification alarmNotification);

	@Query("SELECT * FROM AlarmNotification")
	List<AlarmNotification> getAlarmNotifications();

	@Query("DELETE FROM AlarmNotification WHERE identifier = :identifier")
	void deleteAlarmNotification(String identifier);

	@Query("DELETE FROM AlarmNotification")
	void clear();
}

File: app-android/tutashared/src/main/java/de/tutao/tutashared/data/AlarmInfoDao.kt
package de.tutao.tutashared.data

import androidx.room.Dao
import androidx.room.Insert
import androidx.room.OnConflictStrategy
import androidx.room.Query
import de.tutao.tutashared.alarms.EncryptedAlarmNotificationEntity


@Dao
interface AlarmInfoDao {
	@Insert(onConflict = OnConflictStrategy.REPLACE)
	fun insertAlarmNotification(alarmNotification: EncryptedAlarmNotificationEntity)

	@Query("SELECT * FROM AlarmNotification")
	fun alarmNotifications(): List<EncryptedAlarmNotificationEntity>

	@Query("DELETE FROM AlarmNotification WHERE identifier = :identifier")
	fun deleteAlarmNotification(identifier: String)

	@Query("DELETE FROM AlarmNotification")
	fun clear()
}